Gave the headings to all the files functions

ibilalkayy opened the pull request:
Hey @monstermunchkin

The [previous PR](https://github.com/canonical/lxd/pull/12104) was closed accidentally, so I sent this new one.

Here it contains all the files with headings in their functions and there is no merge conflict also.

Thank you!

monstermunchkin commented:
@ibilalkayy Please rebase correctly, so that you don't need the last two commits.

ibilalkayy commented:
The `lxd/rbac/server.go` file function was added in the `lxd/auth/driver_rbac.go` file and 

Similarly in the `lxd/config/generate/lxddoc.go` file, this new code `func writeDocFile(inputYamlPath, outputTxtPath string) error {}` was updated, that's why I removed the old code `func writeDocFile(path string, yaml *doc) error {}` and updated the one new.

ibilalkayy commented:
@monstermunchkin Now they are fine.

ibilalkayy commented on lxd/auth/driver_rbac.go:
```diff
@@ -474,7 +474,12 @@ func (r *rbac) syncPermissions(username string) error {
 func (r *rbac) postResources(updates []rbacResource, removals []string, force bool) error {
+=======
+// Sends resource updates and removals to the API server and handles conflicts by syncing projects.
+func (r *Server) postResources(updates []rbacResource, removals []string, force bool) error {
```
@monstermunchkin  Due to this function, I added another commit. Even after rebasing, it was still giving me an error. I want you to remove this function and add `func (r *rbac) postResources(updates []rbacResource, removals []string, force bool) error {}` that is matched with the LXD updated code.

monstermunchkin replied:
You'll need to manually fix this change.

ibilalkayy reviewed (commented):
@monstermunchkin I want you to review these two changes because I didn't update them and this is the reason that so many tests are failing.

Even after rebasing, the two functions were not updating and they were requiring commit and that's why I added two commits but then you said and I removed those commits and stashed those changes.

monstermunchkin commented:
> @monstermunchkin I want you to review these two changes because I didn't update them and this is the reason that so many tests are failing.
> 
> Even after rebasing, the two functions were not updating and they were requiring commit and that's why I added two commits but then you said and I removed those commits and stashed those changes.

You removed the commits, but left the conflicts in the file. Those conflicts cause LXD not to build due to malformed go code. Once the conflicts are resolved, it will build again. You will need to edit those files to fix the issue.

ibilalkayy commented:
@monstermunchkin I modified the functions in two files that were giving the errors.

ibilalkayy commented:
Although I created the two commits previously but after rebasing, they were not set up according to the updated LXD code. That's why I manually set them again and matched them with the LXD new code with two new commits.

monstermunchkin commented:
> I don't know how to modify the last commits again after adding them.

With `git rebase -i`. That will allow you to edit older commits. [Here's a tutorial](https://www.bryanbraun.com/2019/02/23/editing-a-commit-in-an-interactive-rebase/#the-interactive-rebase).

ibilalkayy commented:
Hey @monstermunchkin now I understood the problem and the problem is that the two files or the two new commits that I added were not present in previous month version. 

You can check the previous month release of the `lxd/lxd/auth` directory but when I upstreamed the code, it added those two new files also.

Although I searched the whole previous commit history but those two files were not present. So it is not a matter of editing the old commits. It's a matter of adding the two new commits and that's why I added them.

ibilalkayy commented:
Now the commits are edited and the number of commits are the previous one.

monstermunchkin commented:
@ibilalkayy The static analysis tests are failing. Please sort out those issues.

ibilalkayy commented:
@monstermunchkin There are some files that need the endpoint and it will take me some time to edit them.

ibilalkayy commented:
Hey @monstermunchkin, I solved the first check by giving the endpoint to headings but now it is giving me another error that is `Tests / System tests (1.18.x, cluster, zfs) (pull_request)`. I don't know where this problem is present

monstermunchkin commented:
> Hey @monstermunchkin, I solved the first check by giving the endpoint to headings but now it is giving me another error that is `Tests / System tests (1.18.x, cluster, zfs) (pull_request)`. I don't know where this problem is present

The zfs test failures are unrelated to this PR. You just need to make sure the static check succeeds.

ibilalkayy commented:
@monstermunchkin All the checks are successfully passed except this ZFS one.

monstermunchkin commented:
@ibilalkayy Please rebase your changes onto main, instead of adding a merge commit.

ibilalkayy commented:
@monstermunchkin Now I have rebased it and the merge conflict is also removed.

tomponline commented:
Hi @ibilalkayy please can you rebase again so that we can ensure the docs test passes, thanks

ibilalkayy commented:
@tomponline, @monstermunchkin  Finally all the checks are passed and there is no merge conflict.

monstermunchkin commented on lxd/endpoints/object.go:
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/canonical/lxd/shared/logger"
 )
 
+// StorageBucketsCreateListener establishes a secure TCP listener for storage bucket communication,
```
```suggestion
// storageBucketsCreateListener establishes a secure TCP listener for storage bucket communication,
```

monstermunchkin commented on lxd/endpoints/pprof.go:
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/canonical/lxd/shared/logger"
 )
 
+// PprofCreateServer creates a standalone HTTP server dedicated to serving Go's pprof debugging requests.
```
```suggestion
// pprofCreateServer creates a standalone HTTP server dedicated to serving Go's pprof debugging requests.
```

ibilalkayy commented:
@monstermunchkin It will take me some time to look at each heading.

ibilalkayy commented:
@monstermunchkin The comments are now corrected and they are changed from upper case to lower case as they are written in the functions.

monstermunchkin commented:
@ibilalkayy ping me once you want me to review this PR again.

ibilalkayy commented:
@monstermunchkin I have changed the function names in the comments from upper case to lower case and all the checks are passed successfully. Now you can review them.

tomponline commented:
This needs a rebase please

ibilalkayy commented:
@monstermunchkin and @tomponline What do I need to change?

monstermunchkin commented:
> @monstermunchkin and @tomponline What do I need to change?

`lxd/daemon.go` has conflicts which need to be solved.

tomponline commented:
This is proving very difficult to review in time to avoid conflicts.
Are you able to reduce the size of the changes to just functions that you feel could benefit from a doc block? 
I'd like to get this PR merged. Thanks

ibilalkayy commented:
@tomponline I didn't understand what you said. Tell me briefly.

tomponline commented:
Please can you rebase this at least so the PR passes tests

tomponline commented:
Hi @ibilalkayy we've found this PR too large to review.

Also we've recently changed to require the Canonical CLA to be signed see https://github.com/canonical/lxd/blob/main/CONTRIBUTING.md#license-and-copyright

Please can open separate PRs, with commits per change restricted to each package you're changing, this should balance out the number of PRs vs the number of commits per PR and allow us to review the changes in a reasonable time.

Thanks
